osu.Game/Database/LegacyScoreExporter.cs

```diff
@@ -3,21 +3,41 @@
 
 #nullable disable
 
+using System.Collections.Generic;
 using System.IO;
 using System.Linq;
 using osu.Framework.Platform;
 using osu.Game.Extensions;
 using osu.Game.Scoring;
+using osu.Game.Utils;
 
 namespace osu.Game.Database
 {
     public class LegacyScoreExporter : LegacyExporter<ScoreInfo>
     {
         protected override string FileExtension => ".osr";
 
+        private readonly Storage exportStorage;
+
         public LegacyScoreExporter(Storage storage)
             : base(storage)
         {
+            exportStorage = storage.GetStorageForDirectory(@"exports");
+        }
+
+        private string GetScoreExportString(ScoreInfo score) => $"{score.GetDisplayString()} ({score.Date.LocalDateTime:yyyy-MM-dd})";
+
+        public override void Export(ScoreInfo score)
+        {
+            string scoreExportTitle = GetScoreExportString(score).GetValidFilename();
+
+            IEnumerable<string> existingExports = exportStorage.GetFiles("", $"{scoreExportTitle}*{FileExtension}");
+
+            string scoreExportFilename = NamingUtils.GetNextBestFilename(existingExports, $"{scoreExportTitle}{FileExtension}");
+            using (var stream = exportStorage.CreateFileSafely(scoreExportFilename))
+                ExportModelTo(score, stream);
+
+            exportStorage.PresentFileExternally(scoreExportFilename);
```

Comment:
Overriding and copy-pasting the entire method to change one filename is a bit excessive and unwarranted. You'd actually want to move `GetScoreExportString()` to the base exporter, make it `virtual` there, defaulting to `item.GetDisplayString().GetValidFilename()`, and override only that part here.